src/mmap_unix.rs

```diff
@@ -74,6 +74,156 @@ impl error::Error for Error {}
 
 pub type Result<T> = result::Result<T, Error>;
 
+/// A factory struct to build `MmapRegion` objects.
+pub struct MmapRegionBuilder<B = ()> {
+    size: usize,
+    prot: i32,
+    flags: i32,
+    file_offset: Option<FileOffset>,
+    raw_ptr: Option<*mut u8>,
+    hugetlbfs: Option<bool>,
+    bitmap: B,
+}
+
+impl<B: Bitmap + NewBitmap> MmapRegionBuilder<B> {
+    /// Create a new MmapRegionBuilder.
+    pub fn new(size: usize) -> Self {
+        MmapRegionBuilder {
+            size,
+            prot: libc::PROT_READ | libc::PROT_WRITE,
+            flags: libc::MAP_ANONYMOUS | libc::MAP_NORESERVE | libc::MAP_PRIVATE,
+            file_offset: None,
+            raw_ptr: None,
+            hugetlbfs: None,
+            bitmap: B::with_len(size),
+        }
+    }
+
+    /// Create the `MmapRegion` object with the specified mmap memory protection flag `prot`.
+    pub fn with_mmap_prot(mut self, prot: i32) -> Self {
+        self.prot = prot;
+        self
+    }
+
+    /// Create the `MmapRegion` object with the specified mmap `flags`.
+    pub fn with_mmap_flags(mut self, flags: i32) -> Self {
+        self.flags = flags;
+        self
+    }
+
+    /// Modify mmap flags of the builder.
+    pub fn set_mmap_flags(mut self, set: i32, clear: i32) -> Self {
```

Comment:
Can we split this in two functions? `set_mmap_flags` and `clear_mmap_flags`. When reading the code that calls `set_mmap_flags` it will not be obvious what the flags represent. We will also need to make clear what is the difference between `set_mmap_flags` and `with_mmap_flags`. Maybe we can name `set_mmap_flags` to `add_mmap_flags` instead because then the name suggests that the previous set flags are not overwritten?

Reply:
When `flags` is initialized to zero, we don't need this interface anymore:)